js/errors.js

```diff
@@ -283,7 +314,58 @@
     NoConnectionError.prototype = Object.create(Error.prototype);
     NoConnectionError.prototype.constructor = NoConnectionError;
 
+    function removePageCondition(path){
+      var newPath;
+      if (path != undefined && path.indexOf("@before") !== -1) {
+        beforerLiteral = path.match(/(@before\([^\)]*\))/)[1];
```

Comment:
You can use regex in replace, so you can just change these remove page and sort functions to just 

`return path.replace(REGEX, "")`